CMakeLists.txt

```diff
@@ -77,95 +61,145 @@ elseif(${CMAKE_BUILD_TARGET} MATCHES "Both")
+    VERSION ${libuvc_VERSION}
+    SOVERSION ${libuvc_VERSION_MAJOR}
+    EXPORT_NAME UVCShared
+  )
   list(APPEND UVC_TARGETS uvc)
+  if(NOT LibUVC_STATIC)
+    add_library(LibUVC::UVC ALIAS uvc)
+  endif()
+  add_library(LibUVC::UVCShared ALIAS uvc)
 endif()
 
 if(BUILD_UVC_STATIC)
   add_library(uvc_static STATIC ${SOURCES})
-  set_target_properties(uvc_static PROPERTIES OUTPUT_NAME uvc)
+  set_target_properties(uvc_static PROPERTIES
+    OUTPUT_NAME uvc
+    EXPORT_NAME UVCStatic
+  )
   list(APPEND UVC_TARGETS uvc_static)
+  add_library(LibUVC::UVCStatic ALIAS uvc_static)
+  if(LibUVC_STATIC)
+    add_library(LibUVC::UVC ALIAS uvc_static)
+  endif()
 endif()
 
 configure_file(include/libuvc/libuvc_config.h.in
-  ${PROJECT_BINARY_DIR}/include/libuvc/libuvc_config.h @ONLY)
+  include/libuvc/libuvc_config.h
+  @ONLY
+)
 
-foreach(target_name ${UVC_TARGETS})
+foreach(target_name IN LISTS UVC_TARGETS)
+  target_include_directories(${target_name}
+    PUBLIC
+      $<INSTALL_INTERFACE:include>
+      $<BUILD_INTERFACE:${CMAKE_CURRENT_LIST_DIR}/include$<SEMICOLON>>
+      $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}/include>
+  )
+  target_link_libraries(${target_name}
+    PRIVATE LibUSB::LibUSB
+  )
+  if(JPEG_FOUND)
+    target_link_libraries(${target_name}
+      PRIVATE JPEG::JPEG
+    )
+  endif()
   set_target_properties(${target_name} PROPERTIES
-    PUBLIC_HEADER "include/libuvc/libuvc.h;${libuvc_BINARY_DIR}/include/libuvc/libuvc_config.h" )
+      PUBLIC_HEADER "include/libuvc/libuvc.h;${libuvc_BINARY_DIR}/include/libuvc/libuvc_config.h"
+  )
+  if(ENABLE_UVC_DEBUGGING)
+    target_compile_definitions(${target_name}
+      PRIVATE
+        UVC_DEBUGGING
+    )
+  endif()
 endforeach()
 
-if(BUILD_UVC_SHARED)
-  if(JPEG_FOUND)
-    target_link_libraries (uvc ${JPEG_LINK_FLAGS})
-  endif(JPEG_FOUND)
-
-  target_link_libraries(uvc ${LIBUSB_LIBRARIES})
-
-  #add_executable(test src/test.c)
-  #target_link_libraries(test uvc ${LIBUSB_LIBRARIES} opencv_highgui
-  #  opencv_core)
-endif()
-
 if(BUILD_EXAMPLE)
   add_executable(example src/example.c)
-  target_link_libraries(example uvc pthread)
+  find_package(Threads)
+  target_link_libraries(example
+    PRIVATE
+      LibUVC::UVC
+      Threads::Threads
+  )
 endif()
 
 if(BUILD_TEST)
-
-    # Try to find OpenCV using a module or pkg-config. If that doesn't work, search for the header.
-  find_package(opencv QUIET)
-  if(OPENCV_FOUND)
-    set(OPENCV_LINK_FLAGS ${OPENCV_LIBRARIES})
-  else()
-    pkg_check_modules(OPENCV QUIET opencv)
-    if(OPENCV_FOUND)
-        set(OPENCV_INCLUDE_DIR ${OPENCV_INCLUDE_DIRS})
-        set(OPENCV_LINK_FLAGS ${OPENCV_LDFLAGS})
-    endif()
-  endif()
+  find_package(OpenCV
+    REQUIRED COMPONENTS
+      opencv_core
+      opencv_highgui
+  )
 
   add_executable(uvc_test src/test.c)
-  target_link_libraries(uvc_test uvc ${OPENCV_LDFLAGS})
+  target_link_libraries(uvc_test
+    PRIVATE
+      LibUVC::UVC
+      opencv_core
```

Comment:
OpenCV defines targets with transitive dependencies not with namespaces but using `opencv_` prefix. This targets provide necessary include directories and linked flags.